osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/application/config/MessagingConfig.java

```diff
@@ -0,0 +1,116 @@
+    }
+
+    // === JMS SETTINGS IEC60870 REQUESTS ===
+    @Bean
+    public DeviceRequestMessageListener iec60870RequestsMessageListener() {
+        return new DeviceRequestMessageListener();
+    }
+
+    @Bean
+    public JmsConfiguration iec60870RequestJmsConfiguration(final JmsConfigurationFactory jmsConfigurationFactory) {
+        return jmsConfigurationFactory.initializeReceiveConfiguration("jms.iec60870.requests",
+                this.iec60870RequestsMessageListener());
+    }
+
+    @Bean
+    public DefaultMessageListenerContainer iec60870RequestsMessageListenerContainer(
+            final JmsConfiguration iec60870RequestJmsConfiguration) {
+        final DefaultMessageListenerContainer messageListenerContainer = iec60870RequestJmsConfiguration
+                .getMessageListenerContainer();
+
+        // Setting ErrorHandler to prevent logging at WARN level
+        // when JMSException is thrown: Execution of JMS message
+        // listener failed, and no ErrorHandler has been set.
+        messageListenerContainer.setErrorHandler(
+                t -> LOGGER.debug("iec60870RequestsMessageListenerContainer.ErrorHandler.handleError()", t));
+        return messageListenerContainer;
+    }
+
+    @Bean
+    public int maxRedeliveriesForIec60870Requests(final JmsConfiguration iec60870RequestJmsConfiguration) {
+        return iec60870RequestJmsConfiguration.getRedeliveryPolicy().getMaximumRedeliveries();
+    }
+
+    // === JMS SETTINGS: IEC60870 RESPONSES ===
+
+    @Bean
+    public JmsConfiguration iec60870ResponseJmsConfiguration(final JmsConfigurationFactory jmsConfigurationFactory) {
+        return jmsConfigurationFactory.initializeConfiguration("jms.iec60870.responses");
+    }
+
+    @Bean
+    public JmsTemplate iec60870ResponsesJmsTemplate(final JmsConfiguration iec60870ResponseJmsConfiguration) {
+        return iec60870ResponseJmsConfiguration.getJmsTemplate();
+    }
+
+    @Bean
+    public DeviceResponseMessageSender iec60870ResponseMessageSender() {
+        return new DeviceResponseMessageSender();
+    }
+
+    // === JMS SETTINGS: IEC60870 LOG ITEM REQUESTS ===
+
+    @Bean
+    public JmsConfiguration logItemRequestJmsConfiguration(final JmsConfigurationFactory jmsConfigurationFactory) {
+        return jmsConfigurationFactory.initializeConfiguration("jms.iec60870.log.item.requests");
+    }
+
+    @Bean
+    public JmsTemplate logItemRequestsJmsTemplate(final JmsConfiguration logItemRequestJmsConfiguration) {
+        return logItemRequestJmsConfiguration.getJmsTemplate();
+    }
+
+    @Bean
+    public LogItemRequestMessageSender logItemRequestMessageSender() {
+        return new LogItemRequestMessageSender();
+    }
+
+    // === DEVICE MESSAGE LOGGING ===
+
+    @Bean
+    public DeviceMessageLoggingService deviceMessageLoggingService() {
+        return new DeviceMessageLoggingService(this.logItemRequestMessageSender());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this instantiation of "DeviceMessageLoggingService". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS2440)